lib/Cake/Test/Case/Model/Datasource/Database/MysqlTest.php

```diff
@@ -2625,10 +2633,20 @@ public function testArrayConditionsOneElement() {
 		$expected = " WHERE id = (1)";
 		$this->assertEquals($expected, $result);
 
+		$conditions = array('key' => array('939'));
+		$result = $this->Dbo->conditions($conditions);
+		$expected = " WHERE `key` IN ('939')";
```

Comment:
This could also conceivably be converted into `WHERE key = '939'`. That might make the solution more complex though.

Reply:
@tersmitten Sorry, I didn't notice your pull request.

> Is $quoteValues ever false?

It looks like a bug. The `if` check should be removed. The `$quoteValues` argument seems to be no longer used as of [this commit](https://github.com/cakephp/cakephp/commit/39feb3f7c1675f73c15db885ef4cc571428a9520#diff-d61b91ae3172c2a361c5c376c892c195), 

> When does $this->value return an array?

For example, [this call](https://github.com/cakephp/cakephp/blob/2032fef772abcf6d827ba897e620de2b0253871a/lib/Cake/Model/Datasource/DboSource.php#L2904) returns an array. But I think it is a bad idea. No one would not expect that the method returning array. However, since it is a public method, we cannot change the behavior. Instead, having a new `DboSource::val()` not returning array would be an idea. Actually, I was planning to add this method to the core in the past.

Reply:
Adding a new config option to DboSource class may be better.